eth-contracts/test/governance.test.js

```diff
@@ -0,0 +1,468 @@
+      assert.equal(fromBn(txParsedEvent0.event.args.callValue), callValue, 'Expected same txParsedEvent0.event.args.callValue')
+      assert.equal(txParsedEvent0.event.args.signature, signature, 'Expected same txParsedEvent0.event.args.signature')
+      assert.equal(txParsedEvent0.event.args.callData, callData, 'Expected same txParsedEvent0.event.args.callData')
+      assert.equal(txParsedEvent0.event.args.returnData, returnData, 'Expected same txParsedEvent0.event.args.returnData')
+      assert.equal(txParsedEvent1.event.name, 'ProposalOutcomeEvaluated', 'Expected same event name')
+      assert.equal(parseInt(txParsedEvent1.event.args.proposalId), proposalId, 'Expected same event.args.proposalId')
+      assert.equal(parseInt(txParsedEvent1.event.args.outcome), outcome, 'Expected same event.args.outcome')
+      assert.equal(parseInt(txParsedEvent1.event.args.voteMagnitudeYes), fromBn(defaultStakeAmount), 'Expected same event.args.voteMagnitudeYes')
+      assert.equal(parseInt(txParsedEvent1.event.args.voteMagnitudeNo), 0, 'Expected same event.args.voteMagnitudeNo')
+      assert.equal(parseInt(txParsedEvent1.event.args.numVotes), 1, 'Expected same event.args.numVotes')
+
+      // Call getProposalById() and confirm same values
+      const proposal = await governanceContract.getProposalById.call(proposalId)
+      assert.equal(parseInt(proposal.proposalId), proposalId, 'Expected same proposalId')
+      assert.equal(proposal.proposer, proposerAddress, 'Expected same proposer')
+      assert.isTrue(parseInt(proposal.startBlockNumber) > lastBlock, 'Expected startBlockNumber > lastBlock')
+      assert.equal(_lib.toStr(proposal.targetContractRegistryKey), _lib.toStr(targetContractRegistryKey), 'Expected same proposal.targetContractRegistryKey')
+      assert.equal(proposal.targetContractAddress, targetContractAddress, 'Expected same proposal.targetContractAddress')
+      assert.equal(fromBn(proposal.callValue), callValue, 'Expected same proposal.callValue')
+      assert.equal(proposal.signature, signature, 'Expected same proposal.signature')
+      assert.equal(proposal.callData, callData, 'Expected same proposal.callData')
+      assert.equal(proposal.outcome, outcome, 'Expected same outcome')
+      assert.equal(parseInt(proposal.voteMagnitudeYes), defaultStakeAmount, 'Expected same voteMagnitudeYes')
+      assert.equal(parseInt(proposal.voteMagnitudeNo), 0, 'Expected same voteMagnitudeNo')
+      assert.equal(parseInt(proposal.numVotes), 1, 'Expected same numVotes')
+
+      // Confirm all vote states - Vote.No for Voter, Vote.None for all others
+      for (const account of accounts) {
+        const voterVote = await governanceContract.getVoteByProposalAndVoter.call(proposalId, account)
+        if (account == voterAddress) {
+          assert.equal(voterVote, vote)
+        } else {
+          assert.equal(voterVote, defaultVote)
+        }
+      }
+
+      // Confirm Slash action succeeded by checking new Stake values
```

Comment:
validating burn event occurred by checking diff in total supply?

Reply:
i see this is covered in staking.test.js, but seems like it is valid here too - non blocker

Reply:
fair, done